src/router.rs

```diff
@@ -190,18 +208,26 @@ impl Router {
             url = Url::from_generic_url(generic_url).unwrap();
         }
 
-        self.recognize(&req.method, &path).and(
-            Some(IronError::new(TrailingSlash,
+        self.recognize(&req.method, &path).ok().and(
+            Some(IronError::new(RouterError::TrailingSlash,
                                 (status::MovedPermanently, Redirect(url))))
         )
     }
 
     fn handle_method(&self, req: &mut Request, path: &str) -> Option<IronResult<Response>> {
```

Comment:
I don't think this nested typesig is necessary. We can directly return IronResult here, so if we previously returned None we just return RouterError::NotFound inside of an IronError.